python/mxnet/gluon/contrib/estimator/event_handler.py

```diff
@@ -241,17 +241,16 @@ class LoggingHandler(TrainBegin, TrainEnd, EpochBegin, EpochEnd, BatchBegin, Bat
     """
 
     LOG_PER_EPOCH = 1
-    LOG_PER_BATCH = 2
-    LOG_PER_INTERVAL = 3
+    LOG_PER_INTERVAL = 2
 
     def __init__(self, verbose=LOG_PER_EPOCH,
                  train_metrics=None,
                  val_metrics=None,
                  log_interval=1):
```

Comment:
Do you think it would make sense to drop the `verbose` argument completely? For example, if `log_interval=None`, we only log per epoch. If log_interval is set to any positive integer, we log according to that interval?
For example, the following constructor:
```
def __init__(self, log_interval=None,
                 train_metrics=None,
                 val_metrics=None):
```

Reply:
It can be done in this way, but I think verbose is more user friendly. If we use `log_interval=None` in the logging handler, user may need to understand this hack from the documentation. I think we should leave verbose here.

Reply:
`log_interval='epoch' `?

Reply:
Then the type of the variable will be confusing. In the comments of `LoggingHandler`, we have `log_interval: int, default 1`. If we assign a value other than integer type, user may complain about it.

Even though there is no type declaration in python, I believe it is still convention that a variable should only be of one type. It is also beneficial for maintenance.

Reply:
My concern is that having constants
``` python 
LOG_PER_EPOCH = 1
LOG_PER_INTERVAL = 2
```
is not very pythonic. For example users will need to specify

``` python 
est = Estimator(Estimator.LOG_PER_INTERVAL, log_interval=20, train_metrics=X)
```

users may easily by mistake specify

``` python 
est = Estimator(log_interval=20, train_metrics=X)
```

and wonder about why things don't work.

Instead, it's much simpler to 

``` python 
est = Estimator(log_interval='epoch', train_metrics=X)
est = Estimator(log_interval=20, train_metrics=X)
```

or

 ``` python 
est = Estimator('epoch', train_metrics=X)
est = Estimator(20, train_metrics=X)
```

Having variable type arguments is fine in Python / in this case IMO.

Reply:
I see. I'll remove the verbose part and add some documentation to address this issue.